config/routes.rb

```diff
@@ -668,6 +653,73 @@ def route_actions_hash
     actions: LEGACY_CRUD_ACTIONS - [:destroy] + [:show_past]
   )
 
+  # ----- Herbaria: standard actions -------------------------------------------
+  namespace :herbaria do
+    resources :curator_requests, only: [:new, :create]
+    resources :curators, only: [:create, :destroy], id: /\d+/
+    resources :merges, only: [:create]
+    resources :nexts, only: [:show], id: /\d+/
+  end
+  resources :herbaria, id: /\d+/
+  # Herbaria: standard redirects of Herbarium legacy actions
+  redirect_legacy_actions(
+    old_controller: "herbarium", new_controller: "herbaria",
+    actions: LEGACY_CRUD_ACTIONS - [:controller, :index, :show_past]
+  )
+  # Herbaria: non-standard redirects of legacy Herbarium actions
+  # Rails routes currently accept only template tokens
+  # rubocop:disable Style/FormatStringToken
+  get("/herbarium/herbarium_search", to: redirect(path: "herbaria"))
+  get("/herbarium/index", to: redirect(path: "herbaria"))
+  get("/herbarium/list_herbaria", to: redirect(path: "herbaria?flavor=all"))
+  get("/herbarium/request_to_be_curator/:id",
+      to: redirect(path: "herbaria/curator_requests/new?id=%{id}"))
+
+  # Herbaria: complicated redirects of legacy Herbarium actions
+  # Actions needing two routes in order to successfully redirect
+  #
+  # The immediately following "match" and "get" combine to redirect
+  # the legacy herbarium/delete_curator to the new herbaria/curators
+  # The "match" redirects
+  #   GET("/herbarium/delete_curator/nnn?user=uuu") and
+  #   POST("/herbarium/delete_curator/nnn?user=uuu")
+  # to
+  #   GET("/herbaria/curators/nnn?user=uuu")
+  # which would throw: No route matches [GET] "/herbaria/curators/nnnnn"
+  # absent the following get
+  match("/herbarium/delete_curator/:id",
+        to: redirect(path: "/herbaria/curators/%{id}"),
+        via: [:get, :post])
+  get("/herbaria/curators/:id", to: "herbaria/curators#destroy", id: /\d+/)
+
+  # The next post and get combine to redirect the legacy
+  #   POST /herbarium/request_to_be_curator to
+  #   POST herbaria/curator_requests#create
+  post("/herbarium/request_to_be_curator/:id",
+       to: redirect(path: "/herbaria/curator_requests?id=%{id}"))
+  get("/herbaria/curator_requests",
+      to: "herbaria/curator_requests#create", id: /\d+/)
+
+  # The next post and get combine to redirect the legacy
+  #   POST /herbarium/show_herbarium/:id to
+  #   POST herbaria/curators#create
+  post("/herbarium/show_herbarium", to: redirect(path: "herbaria/curators"))
+  get("/herbaria/curators", to: "herbaria/curators#create", id: /\d+/)
+
+  # Herbaria: broken redirects of legacy Herbarium actions
+  get("/herbarium/next_herbarium/:id",
+      to: redirect(path: "herbaria/%{id}?flow=next"))
+  get("/herbarium/prev_herbarium/:id",
+      to: redirect(path: "herbaria/%{id}?flow=prev"))
+  get("/herbarium/merge_herbaria",
+      to: redirect(path: "herbaria/merges#create"), that: /\d+/, this: /\d+/)
```

Comment:
### These redirects are broken. See test output below.
### `next` and `prev`
I can't figure out how to add a new, static parameter while retaining the `q` param.
Potential solutions:
- My preferred solution: Don't redirect the legacy action. I don't think this will affect anyone. There shouldn't be old links in the cloud that point to the legacy `next` and `prev` actions
- 2nd choice: A namespaced controller with a `show` action and a param to control direction. Somehow the analogous routes work for a separate controller, e.g.
```ruby
  get("/herbarium/next_herbarium/:id",
      to: redirect(path: "herbaria/nexts/%{id}?next=next"))
```
- 3rd choice: Separate non-CRUD `next` and `prev` actions in HerbariaController. 
### merge
I can't figure out how to route the legacy GET action to a new POST action.
Potential solutions:
- My preferred solution: Don't redirect the legacy action. There shouldn't be a link in the cloud that points to a merger.
- 2nd choice: Use `new` instead of `create`. This works, but has security implications; Rails advises _never_ writing to the db from a GET. See https://guides.rubyonrails.org/v5.2/routing.html#http-verb-constraints
-----
```sh
 FAIL["test_next_herbarium", #<Minitest::Reporters::Suite:0x000055613a15f5b8 @name="RedirectsTest">, 8.719991796999238]
 test_next_herbarium#RedirectsTest (8.72s)
        --- expected
        +++ actual
        @@ -1 +1 @@
        -"/herbaria/1022913654?flow=next&q=1Wi"
        +"/herbaria/1022913654?flow=next?q=1Wi"
        test/integration/redirects_test.rb:22:in `assert_old_url_redirects_to_new_path'
        test/integration/redirects_test.rb:291:in `test_next_herbarium'

 FAIL["test_prev_herbarium", #<Minitest::Reporters::Suite:0x00007fecec57de48 @name="RedirectsTest">, 12.29093096699944]
 test_prev_herbarium#RedirectsTest (12.29s)
        --- expected
        +++ actual
        @@ -1 +1 @@
        -"/herbaria/1022913654?flow=prev&q=1Wl"
        +"/herbaria/735604515?flow=prev?q=1Wl"
        test/integration/redirects_test.rb:22:in `assert_old_url_redirects_to_new_path'
        test/integration/redirects_test.rb:308:in `test_prev_herbarium'
```

Reply:
I think it's perfectly reasonable in these cases not to redirect the legacy action.  As you say, it's very unlikely that these have been saved as links any where and even if they are I doubt anyone will complain about it.